at-dungnguyen/Final/main.rb

```diff
@@ -0,0 +1,279 @@
+      p 'Login Success'
+    else
+      p 'Wrong username or password'
+    end
+  end
+
+  def register
+    p 'Name: '
+    name = gets.chomp
+    username, password = input_account
+    clear
+    if @sc.register(name, username, password)
+      p 'Register Success'
+    else
+      p 'Username Exist!'
+    end
+  end
+
+  def forgot_password
+    p 'Input username:'
+    username = gets.chomp
+    @sc.forgot_password username
+  end
+
+  def input_account
+    p 'Username:'
+    username = gets.chomp
+    p 'Password:'
+    password = STDIN.noecho(&:gets).chomp
+    [username, password]
+  end
+
+  # Team
+  def team_management
+    p 'Team Management'
+    p '==============='
+    p '1. Show list of members'
+    p '2. Back'
+    p '3. Logout'
+    p '4. Add a member' if @user['role'] == SessionController::LEADER
+    p '5. Create new team' if @user['role'] == SessionController::ADMIN
+    p '0. Exit'
+  end
+
+  def execute_team(function)
+    case function
+    when 1
+      members = @tc.members(@user['team'])
+      if members.any?
+        members.each do |mem|
+          p "ID: #{mem['id']}, Name: #{mem['name']}"
+        end
+      else
+        p 'No result'
+      end
+      p '====================================================================='
+    when 2
+      @flag_team = false
+    when 3
+      @flag_team = false
+      @user = nil
+    when 4
+      @tc.add_member(@user['team']) if @user['role'] == SessionController::LEADER
+    when 5
+      @tc.add_team if @user['role'] == SessionController::ADMIN
+    when 0
+      @flag_team = false
+      execute2(0)
+    end
+  end
+
+  # Norification
+  def notification_management
+    p 'Notification Management'
+    p '======================='
+    p '1. List Notification'
+    p '2. Read a notification'
+    p '3. Back'
+    p '4. Logout'
+    p '5. Send a notification' if @user['role'] == SessionController::LEADER
+    p '6. Delete a notification' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_notify(function)
+    case function
+    when 1
+      all_notify
+    when 2
+      g_notify
+    when 3
+      @flag_notify = false
+    when 4
+      @flag_notify = false
+      @user = nil
+    when 5
+      send_notify
+    when 6
+      del_notify
+    when 0
+      @flag_notify = false
+      execute2(0)
+    end
+  end
+
+  def all_notify
+    list_notify = @nc.get_items(@user['team'])
+    if list_notify.any?
+      list_notify.each do |notify|
+        puts "#{notify['title']} | #{notify['publish_date']}"
+      end
+    end
+    p '====================================================================='
+  end
+
+  def g_notify
+    p 'Input id notification'
+    id = input
+    notify = @nc.get_item(id)
+    p "#{notify['title']} | #{notify['publish_date']}"
+    p notify['content']
+  end
+
+  def send_notify
+    p 'Input title'
+    title = gets.chomp
+    p 'Input content'
+    content = gets.chomp
+    publish_date = Time.now
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/59f7885620a5f3c551417f8e71806b1b95622bf8/at-dungnguyen/Final/main.rb#L170"></span>Rails/TimeZone: Do not use <code>Time.now</code> without zone. Use one of <code>Time.zone.now</code>, <code>Time.current</code>, <code>Time.now.in_time_zone</code>, <code>Time.now.utc</code>, <code>Time.now.getlocal</code>, <code>Time.now.xmlschema</code>, <code>Time.now.iso8601</code>, <code>Time.now.jisx0301</code>, <code>Time.now.rfc3339</code>, <code>Time.now.httpdate</code>, <code>Time.now.to_i</code>, <code>Time.now.to_f</code> instead. (https://github.com/rubocop-hq/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)</td>
    </tr>
  </tbody>
</table>